src/python/grpcio_tests/tests_aio/unit/interceptor_test.py

```diff
@@ -573,6 +573,100 @@ class Interceptor(aio.UnaryUnaryClientInterceptor):
 
             self.assertEqual(await call.code(), grpc.StatusCode.OK)
 
+    async def test_add_done_callback_before_finishes(self):
+        called = False
+        interceptor_can_continue = asyncio.Event()
+
+        def callback(call):
+            nonlocal called
+            called = True
+
+        class Interceptor(aio.UnaryUnaryClientInterceptor):
+
+            async def intercept_unary_unary(self, continuation,
+                                            client_call_details, request):
+
+                await interceptor_can_continue.wait()
+                call = await continuation(client_call_details, request)
+                return call
+
+        async with aio.insecure_channel(self._server_target,
+                                        interceptors=[Interceptor()
+                                                     ]) as channel:
+
+            multicallable = channel.unary_unary(
+                '/grpc.testing.TestService/UnaryCall',
+                request_serializer=messages_pb2.SimpleRequest.SerializeToString,
+                response_deserializer=messages_pb2.SimpleResponse.FromString)
+            call = multicallable(messages_pb2.SimpleRequest())
+            call.add_done_callback(callback)
```

Comment:
The callback's execution order is not guaranteed by the AsyncIO API. We might want to change `called` to a `asyncio.Event()`.

Reply:
Good point! changed.